lib/remoteproc/remoteproc.c

```diff
@@ -36,11 +36,12 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 {
 	struct metal_list *node;
 	struct remoteproc_mem *mem;
+	unsigned int str_size;
 
 	metal_list_for_each(&rproc->mems, node) {
 		mem = metal_container_of(node, struct remoteproc_mem, node);
 		if (name) {
-			if (!strncmp(name, mem->name, sizeof(mem->name)))
+			if (!strcmp(name, mem->name))
```

Comment:
This change is problematic, since it is currently possible that mem->name is not null-terminated. Linux remoteproc solves this by forcibly null-terminating the string, possibly truncating it. strncmp() is safer here, because it will terminate without running past the end of mem->name.

Reply:
The issue here seems not mem->name but "name" parameter. The name size is considered as unknown by MISRA. this generates a High level error:
  Overrun in strncmp with string `mem->name'. Length passed to strncmp is 32, but string is of length <unknown size>	LIB-strncmp-overrun	High	openamp\open-amp\lib\remoteproc\remoteproc.c:44
    remoteproc_get_mem	openamp\open-amp\lib\remoteproc\remoteproc.c
      if (name) is true	openamp\open-amp\lib\remoteproc\remoteproc.c:43
      if (!strncmp(name, mem->name, sizeof((mem->name))))	openamp\open-amp\lib\remoteproc\remoteproc.c:44
      LIB-strncmp-overrun	openamp\open-amp\lib\remoteproc\remoteproc.c:44
 
Following implementation matches with MISRA compliance. Could you tell me if this should be ok for you?
  if (name && strlen(name) <= sizeof(mem->name))  {
       if (!strncmp(name, mem->name, strlen(name)))
		return mem;

Reply:
I believe that the error reported here is erroneous:) strncmp() will compare at most sizeof(mem->name) bytes, stopping as soon as it sees a null terminator in either string.

That being said, I would have no objection to your alternative implementation, as long as you also add a comment explaining why the code is there. Otherwise, we risk someone removing it as unnecessary in the future.

Reply:
As CPU can access to a unexpected memory region, i understand the report. Now it is a read access could not be a High severity report... But anyway, i'm not 100% happy with my fix which is more a compromize than an elegant solution.  I also notice that there is no check on name parameter size... I 'm trying to find a alternative...